scripts/spm-integration-test.sh

```diff
@@ -0,0 +1,90 @@
+#!/bin/bash
+
+set -e -uxf -o pipefail
+
+# Function to check if a service is healthy
+check_service_health() {
+  local service_name=$1
+  local url=$2
+  local retry_count=10
+  local wait_seconds=5
+
+  echo "Checking health of service: $service_name at $url"
+
+  for i in $(seq 1 $retry_count); do
+    if curl -s -L --head --request GET "$url" | grep "200 OK" > /dev/null; then
+      echo "$service_name is healthy"
+      return 0
+    else
+      echo "Waiting for $service_name to be healthy... ($i/$retry_count)"
+      sleep $wait_seconds
+    fi
+  done
+
+  echo "Error: $service_name did not become healthy in time"
+  return 1
+}
+
+# Function to check if all services are healthy
+wait_for_services() {
+  echo "Waiting for services to be up and running..."
+  check_service_health "Jaeger" "http://localhost:16686"
+  check_service_health "Prometheus" "http://localhost:9090/graph"
+  check_service_health "Grafana" "http://localhost:3000"
+}
+# Function to check SPM
+check_spm(){
+  echo "Checking spm"
+  services_list=("driver" "customer" "mysql" "redis" "frontend" "route" "ui" )
+  echo "Waiting for 60 seconds for services to start"
+  sleep 60
```

Comment:
We do not use unconditional waiting, it slows down CI

Reply:
it takes time to create spans for services otherwise the metrics in response will be empty

Reply:
Maybe keep querying until you see some metrics in the response and timeout after a particular set interval ? Is this a better option

Reply:
Yes. Usually in the integration tests we keep checking until a satisfactory condition is met or a certain timeout is reached (to account for delays in start-ups, but in this case also for delays in data processing).

I was thinking that we want to loop until we have at least 3 data points available for each service. I updated the ticket with more details.